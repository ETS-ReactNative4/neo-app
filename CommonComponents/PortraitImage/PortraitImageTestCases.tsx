import React from "react";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import PortraitImage from "./PortraitImage";

const PortraitImageTestCases: ITestCase[] = [
  {
    title: "Portrait Image",
    Component: (
      <PortraitImage
        imageSource={"https://d3lf10b5gahyby.cloudfront.net/city/paris.jpg"}
      />
    )
  }
];

export default PortraitImageTestCases;
