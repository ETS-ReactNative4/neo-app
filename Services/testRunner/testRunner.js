import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
