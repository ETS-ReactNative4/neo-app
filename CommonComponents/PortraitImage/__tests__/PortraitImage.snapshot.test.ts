import PortraitImageTestCases from "../PortraitImageTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

PortraitImageTestCases.map(testRunner);
