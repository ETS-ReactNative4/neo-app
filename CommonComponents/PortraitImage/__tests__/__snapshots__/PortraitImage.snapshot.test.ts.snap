// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portrait Image 1`] = `
<View>
  <Image
    onError={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://d3lf10b5gahyby.cloudfront.net/city/paris.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 4,
        "height": 200,
        "width": 146,
      }
    }
  />
</View>
`;
